packages/smart/contracts/rewards/MasterChef.sol

```diff
@@ -522,7 +557,21 @@ contract MasterChef is OpenZeppelinOwnable.Ownable {
         RewardPoolLookupInfo memory _rewardPoolLookupInfo =
             rewardPoolLookup[address(_ammFactory)][address(_marketFactory)][_marketId];
 
-        require(_rewardPoolLookupInfo.created, "Reward pool has not been created.");
+        uint256 _pid = _rewardPoolLookupInfo.pid;
+
+        // If not created should attempt to create it.
+        if (!_rewardPoolLookupInfo.created) {
+            BPool _bPool = _ammFactory.getPool(_marketFactory, _marketId);
+            require(_bPool != BPool(0), "Pool not created.");
```

Comment:
Will people be able to make pools with the AMMFactory directly? It looks like if they do then the pool will forever be ineligible for rewards.